src/controllers/book.controller.ts

```diff
@@ -2,21 +2,29 @@ import { Request, Response } from "express";
 import { SortOptions } from "../types/req.filter";
 import { RegisterBookSchema, UpdateBookSchema } from "../models/book.models";
 import bookService from "../services/book.service";
+import imageService from "../services/image.service";
 import { BaseController } from "../types/base.controller";
+
 class BookController extends BaseController {
   /**
    * It validates the request body with the RegisterBookSchema, then calls the bookService.createBook function, and finally sends the response
    * @param {Request | any} req
    * @param {Response} res
    */
-  async createBook(req: Request | any, res: Response) {
+  async createBook(req: Request | any, res: Response): Promise<void> {
     try {
-      const data = await RegisterBookSchema.validateAsync(req.body);
+      const data = await RegisterBookSchema.validateAsync(req.body)
+      if (req.file) {
```

Comment:
This logic should be inside the service, controller has single responsibility to handle request